addons/l10n_anz_ubl_pint/models/account_edi_xml_pint_anz.py

```diff
@@ -0,0 +1,116 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, _
+
+ANZ_TAX_CATEGORIES = {'S', 'E', 'Z', 'G', 'O'}
+
+
+class AccountEdiXmlUBLPINTANZ(models.AbstractModel):
+    _inherit = "account.edi.xml.ubl_bis3"
+    _name = "account.edi.xml.pint_anz"
+    _description = "Australia & New Zealand implementation of Peppol International (PINT) model for Billing"
+    """
+    Pint is a standard for International Billing from Peppol. It is based on Peppol BIS Billing 3.
+    It serves as a base for per-country specialization, while keeping a standard core for data being used
+    across countries. This is not meant to be used directly, but rather to be extended by country-specific modules.
+
+    The ANZ PINT format is the Australia & New Zealand implementation of PINT.
+
+    * PINT Official documentation: https://docs.peppol.eu/poac/pint/pint/
+    * PINT ANZ Official documentation: https://docs.peppol.eu/poac/aunz/pint-aunz/
+    """
+
+    def _export_invoice_filename(self, invoice):
+        # EXTENDS account_edi_ubl_cii
+        return f"{invoice.name.replace('/', '_')}_pint_anz.xml"
+
+    def _get_partner_party_vals(self, partner, role):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals = super()._get_partner_party_vals(partner, role)
+
+        for party_tax_scheme in vals['party_tax_scheme_vals']:
+            party_tax_scheme['tax_scheme_vals'] = {'id': 'GST'}
+
+        return vals
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+        company_currency = invoice.company_id.currency_id
+        if invoice.currency_id != company_currency:
+            # if company currency != invoice currency, need to add a TaxTotal section
+            # see https://docs.peppol.eu/poac/aunz/pint-aunz/bis/#_tax_in_accounting_currency
+            tax_totals_vals = {
+                'currency': company_currency,
+                'currency_dp': company_currency.decimal_places,
+                'tax_amount': taxes_vals['tax_amount'],
+            }
+            vals_list.append(tax_totals_vals)
+        return vals_list
+
+    def _get_tax_unece_codes(self, invoice, tax):
+        """ The GST category must be provided in the file.
+        See https://docs.peppol.eu/poac/aunz/pint-aunz/bis/#_tax_category_code
+        """
+        # OVERRIDE account_edi_ubl_cii
+        supplier = invoice.company_id.partner_id.commercial_partner_id
+        if not supplier.vat:
+            return 'O'  # a business is not registered for GST, the business cannot issue tax invoices. In this case, the GST category code should be O (Outside scope of tax).
+        return super()._get_tax_unece_codes(invoice, tax)
```

Comment:
This function should return a dictionary (see in account_edi_ubl_cii), otherwise, we'll get tracebacks like `'str' object has no attribute 'get'`